src/test/run-pass/signal-alternate-stack-cleanup.rs

```diff
@@ -31,11 +31,23 @@ extern fn send_signal() {
     }
 }
 
+// cast $a to the type of $b
+macro_rules! as_ {
+    ($a:ident as $b:expr) => (
+        if false { $b } else { $a as _ }
+    )
+}
+
 fn main() {
     unsafe {
         // Install signal hander that runs on alternate signal stack.
         let mut action: sigaction = std::mem::zeroed();
-        action.sa_flags = SA_SIGINFO | SA_ONSTACK;
+        // workaround to build in android/aarch64 and android/x86_64
+        // the intention is:
+        // action.sa_flags = SA_ONSTACK | SA_SIGINFO;
+        // See https://github.com/rust-lang/libc/pull/511
```

Comment:
It this is too much, we can ignore this test in the affected targets.

Reply:
Or you could write this as 

```rust
action.sa_flags = (SA_ONSTACK | SA_SIGINFO) as _;
```

Reply:
Oh, this is embarrassing... Fixed it.